### PR TITLE
fix: Docker registry not configured in GKE

### LINKED ISSUE #3857: Docker registry not configured in GKE when installing with `jx install..`
### Summary 
I am installing Jenkins X on a previously created cluster in GKE with terraform (generated with `jx create terraform`, and  docker-registry is not configured, so a quickstart execution gets stuck without running the pipeline.

If you create the same cluster from scratch using `jx create cluster gke ...` it works.

### Steps to reproduce the behavior
- Create terraform scripts and plan with `jx create terraform`
- Install Jenkins X (Tekton) with `jx install...`:
```
$ jx install --provider gke \
> --default-admin-password $MYPWD \
> --default-environment-prefix $JX_CLUSTER \
> --environment-git-owner $GH_ORG \
> --tekton \
> --no-tiller
```
- Create a Go quickstart with `jx create quickstart -l go`

### Expected behavior
The expected behaviour should be the same like installing with creating the cluster command `jx create cluster gke`. If the cluster was created with the terraform plan generated by Jenkins X, an installation should behave the same as installing with the one lined command.

### Actual behavior
- Cluster is created
- Jenkins X is installed
- But any quickstart is not executing the pipeline because it cannot find the docker registry:
```
$ jx logs pipelinerunner
WARNING: Waiting for a running pod in namespace jx with labels app=pipelinerunner
WARNING: Found newest pod: pipelinerunner-f98fbd455-vwvk8
{"level":"info","msg":"Waiting for dynamic Tekton Pipelines at http://:8080/","time":"2019-05-03T16:24:24Z"}
{"level":"info","msg":"Generated Git credentials file /home/jenkins/git/credentials\n","time":"2019-05-03T16:35:52Z"}
{"level":"info","msg":"triggering pipeline for repo https://github.com/dcanadillas-kube/demo-tf-1.git branch master revision master context \n","time":"2019-05-03T16:35:52Z"}
{"level":"info","msg":"Deleting and cloning the Jenkins X versions repo","time":"2019-05-03T16:35:52Z"}
{"level":"info","msg":"Cloning the Jenkins X versions repo https://github.com/jenkins-x/jenkins-x-versions.git with ref refs/heads/master to /root/.jx/jenkins-x-versions\n","time":"2019-05-03T16:35:52Z"}
Enumerating objects: 1260, done.
Total 1260 (delta 0), reused 0 (delta 0), pack-reused 1260
{"level":"info","msg":"shallow cloning repository https://github.com/dcanadillas-kube/demo-tf-1.git to temp dir /tmp/git290189597\n","time":"2019-05-03T16:35:53Z"}
{"level":"warning","msg":"no SHAs to merge, falling back to initial cloned commit","time":"2019-05-03T16:35:56Z"}
{"level":"info","msg":"setting up docker registry for https://github.com/dcanadillas-kube/demo-tf-1.git\n","time":"2019-05-03T16:35:56Z"}
{"level":"info","msg":"removing the temp directory /tmp/git290189597\n","time":"2019-05-03T16:35:56Z"}
{"level":"error","msg":"Missing option: --docker-registry Missing option: --docker-registry","time":"2019-05-03T16:35:56Z"}
{"level":"error","msg":"Missing option: --docker-registry","time":"2019-05-03T16:35:56Z"}
```

Just realized that the docker registry Config Map is not well configured:
```
$ kubectl get cm jenkins-x-docker-registry -o yaml
apiVersion: v1
data:
  docker.registry: "" #THIS SHOULDN'T BE EMPTY
kind: ConfigMap
metadata:
  annotations:
...
```

So the pipeline doesn't even start

### Jx version

The output of `jx version` is:

```
NAME               VERSION
jx                 2.0.75
jenkins x platform 2.0.191
Kubernetes cluster v1.11.8-gke.6
kubectl            v1.13.3
helm client        Client: v2.11.0+g2e55dbe
git                git version 2.11.0
Operating System   Debian GNU/Linux 9.8 (stretch)
```

### Jenkins type

- [x] Next Generation (Tekton + Prow)
- [ ] Classic Jenkins
- [ ] Serverless Jenkins (JenkinsFileRunner + Prow)

### Kubernetes cluster

Created in GKE with Terraform jx plan:
```
jx create terraform -c $JX_CLUSTER=gke \
--gke-machine-type n1-standard-2 \
--gke-max-num-nodes=5 --gke-min-num-nodes=3 \
--gke-project-id=emea-sa-demo \
--gke-zone=europe-west1-c \
--gke-use-enhanced-apis --gke-use-enhanced-scopes \
--organisation-name=terraform-gke \
--skip-login
```
Then:
```
cd ~/.jx/organisations/organisation-terraform-gke/clusters/dc-cluster/terraform/
terraform init
terraform plan
terraform apply
```

### Operating system / Environment

Google Cloud Shell (Debian based):
```
$ uname -ar && cat /etc/debian_version
Linux cs-6000-devshell-vm-706ffcc3-3ad6-43a5-b6c3-8be9a6dbfcba 4.14.104+ #1 SMP Tue Apr 23 18:16:25 PDT 2019 x86_64 GNU/Linux
9.8
```

### PR DESCRIPTION
**Pull request by carlossg**
when installing with `jx install`

Fixes #3857

Signed-off-by: Carlos Sanchez <carlos@apache.org>

#### Submitter checklist

- [ ] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Which issue this PR fixes

fixes #

### CODE REVIEW

**Comment by dcanadillas** [comment]
@carlossg I've tested locally and now docker registry is configured correctly, so this issue seems to work. But also options.Flags.Kaniko is false by default and needs to be true. So when using `jx install --provider gke ...` seems that the install.go does not configure Kaniko and when trying to build a Quickstart is needed (the secret is not found because it's not installed)

File: pkg/jx/cmd/install.go [hunk 1]
```diff
@@ -515,6 +515,20 @@ func (options *InstallOptions) Run() error {
 		return errors.Wrap(err, "configuring minikube from kubectl context")
 	}
 
+	if options.Flags.DockerRegistryOrg == "" {
```
Comment (ccojocar) [inline pkg/jx/cmd/install.go]: @carlossg I would move this to `checkFlags` method. We try to keep the default flags configuration in a central place. Thanks

**Comment by carlossg** [comment]
there's more stuff that needs to be moved from cluster creation to jx install and I don't have the time to get through all of it and do proper testing

**Comment by deanesmith** [comment]
Any update on this?  I just ran in to it.

**Comment by rawlingsj** [comment]
@deanesmith what version of `jx` do you have?  I think this should have been fixed with this PR 10 days ago https://github.com/jenkins-x/jx/pull/4368.  I think we can close this PR.

**Comment by deanesmith** [comment]
k, thanks. Will delete local cache and try again.

**Pull request closed without merging**